balboa-http/src/main/scala/com/socrata/balboa/server/MainServlet.scala

```diff
@@ -0,0 +1,73 @@
+package com.socrata.balboa.server
+
+import javax.servlet.http.HttpServletRequest
+
+import com.socrata.balboa.BuildInfo
+import com.socrata.balboa.metrics.data.BalboaFastFailCheck
+import com.socrata.balboa.server.rest.{EntitiesRest, MetricsRest}
+import com.typesafe.scalalogging.slf4j.StrictLogging
+import org.json4s.{DefaultFormats, Formats}
+import org.scalatra.{InternalServerError, NotFound, Ok, ScalatraServlet}
+import org.scalatra.json.JacksonJsonSupport
+
+import scala.collection.JavaConverters._
+
+class MainServlet extends ScalatraServlet with StrictLogging with JacksonJsonSupport {
+
+  protected implicit val jsonFormats: Formats = DefaultFormats
+
+  val versionString = BuildInfo.toJson
+
+  def getAccepts(req: HttpServletRequest): Seq[String] = {
+    req.getHeaders("accept").asScala.toSeq
+  }
+
+  get("/*") {
+    contentType = "application/json"
+    NotFound(Error(404, "Not found."))
+  }
+
+  get("/version*") {
+    contentType = "application/json"
+    if (BalboaFastFailCheck.getInstance.isInFailureMode) {
+      InternalServerError
+    } else {
+      Ok(versionString)
+    }
+  }
+
+  get("/entities*") {
+    val response = EntitiesRest(params)
+    contentType = response.contentType
+    response.result
+  }
+
+  get("/metrics/:entityId") {
+    val entityId = params("entityId")
+    val response = MetricsRest.get(entityId, params, getAccepts(request))
+    contentType = response.contentType
+    response.result
+  }
+
+  // Catch the extra paths to match behavior of old API
+  get("/metrics/:entityId/*") {
```

Comment:
Since this method is exactly the same as `get("/metrics/:entityId")`, can you combine them? How about something like this, http://scalatra.org/2.4/guides/http/routes.html#toc_295, or a regex pattern matcher if necessary?